rti/mosaic-rti-core/src/main/java/org/eclipse/mosaic/rti/federation/LocalFederationManagement.java

```diff
@@ -263,10 +264,16 @@ protected void startFederate(FederateDescriptor handle) throws Exception {
         errorLoggingThread.start();
         loggingThreads.put(handle.getId(), errorLoggingThread);
 
+        // FIXME: Omnetpp/Ns3 ambassadors must read from the input stream. As we cannot simply split the stream,
+        //        we need to call connectToFederate before starting the ProcessLoggingThread
+        //
         // call connectToFederateMethod of the current federate an extract
         // possible output from the federates' output stream (e.g. port number...)
         // note: error- and input streams were read in this class now due to conflicting stream access
-        handle.getAmbassador().connectToFederate(LOCALHOST, p.getInputStream(), p.getErrorStream());
+        handle.getAmbassador().connectToFederate(LOCALHOST,
+                new CloseShieldInputStream(p.getInputStream()), // prevent streams from closing by ambassador
```

Comment:
https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/input/CloseShieldInputStream.html

According to this documentation the constructor is deprecated and the class should be used with `wrap()` instead

Reply:
In the version we currently use (and which is approved by Eclipse IP Team) this method is not available yet.